makefiles/suit.v4.inc.mk

```diff
@@ -20,33 +20,40 @@ SUIT_DEVICE_ID ?= $(BOARD)
 #
 # SUIT encryption keys
 #
+
+# Specify key to use.
+# Will use $(SUIT_KEY_DIR)/$(SUIT_KEY) $(SUIT_KEY_DIR)/$(SUIT_KEY).pub as
+# private/public key files, similar to how ssh names its key files.
+SUIT_KEY ?= default
+
 ifeq (1, $(RIOT_CI_BUILD))
   SUIT_KEY_DIR ?= $(BINDIR)
 else
   SUIT_KEY_DIR ?= $(RIOTBASE)/keys
 endif
 
-SUIT_KEY ?= $(SUIT_KEY_DIR)/secret.key
-SUIT_PUB ?= $(SUIT_KEY_DIR)/public.key
+SUIT_SEC ?= $(SUIT_KEY_DIR)/$(SUIT_KEY)
+SUIT_PUB ?= $(SUIT_KEY_DIR)/$(SUIT_KEY).pub
 
 SUIT_PUB_HDR = $(BINDIR)/riotbuild/public_key.h
 SUIT_PUB_HDR_DIR = $(dir $(SUIT_PUB_HDR))
 CFLAGS += -I$(SUIT_PUB_HDR_DIR)
 BUILDDEPS += $(SUIT_PUB_HDR)
 
-$(SUIT_KEY) $(SUIT_PUB):
+$(SUIT_SEC) $(SUIT_PUB):
 	@echo suit: generating key pair in $(SUIT_KEY_DIR)
 	@mkdir -p $(SUIT_KEY_DIR)
-	@$(RIOTBASE)/dist/tools/suit_v4/gen_key.py $(SUIT_KEY) $(SUIT_PUB)
+	@$(RIOTBASE)/dist/tools/suit_v4/gen_key.py $(SUIT_SEC) $(SUIT_PUB)
 
 $(SUIT_PUB_HDR): $(SUIT_PUB)
 	@mkdir -p $(SUIT_PUB_HDR_DIR)
-	@cd $(dir $(SUIT_PUB)) && xxd -i $(notdir $(SUIT_PUB)) > $@
+	@cp $(SUIT_PUB) $(SUIT_PUB_HDR_DIR)/public.key
+	@cd $(SUIT_PUB_HDR_DIR) && xxd -i public.key > $@
```

Comment:
Should be `xxd -i $(notdir $(SUIT_PUB))`

Reply:
Or maybe even `xxd -i $(SUIT_KEY).pub` (and also on the line above)

Reply:
no, the key filename needs to be exactly "public.key", as that is how xxd will name the symbol. That is why ```$(SUIT_PUB)``` is copied to the hdr dir, and why we do "cd" before calling xxd.

Reply:
Ah, ok then. Thanks!